README.md

```diff
@@ -1,70 +1,137 @@
 
-## Using CVO to render the release-image locally
+The Cluster Version Operator continuously fetches information about upgrade
+paths for the configured channel from the OpenShift Update Service (OSUS). It
+stores the recommended update options in the `status.availableUpdates` field of
+its `ClusterVersion` resource.
 
-1. Run the following command to get render the release-payload contents to `/tmp/cvo`
+The intent to upgrade the cluster to another version is expressed by storing the
+desired version in the `spec.desiredUpdate` field, either manually by the
+administrator through the CLI or web console, or CVO does it automatically when
+automated upgrades are enabled.
 
-```sh
-podman run --rm -ti \
-    -v /tmp/cvo:/tmp/cvo:z \
-    <release image> \
-        render \
-        --output-dir=/tmp/cvo \
-        --release-image="<release image>"
-```
+When `spec.desiredUpdate` does not match the current cluster version, CVO will
+start updating the cluster. It downloads the release payload image, validates
+it, and [systematically reconciles](https://github.com/openshift/enhancements/blob/master/dev-guide/cluster-version-operator/user/reconciliation.md)
+the Cluster Operator resources to match the updated manifests delivered
+in the release payload image.
 
-`<release image>` can be personal release image generated using [this](#building-release-image-using-local-cvo) or Origin's release image like `registry.ci.openshift.org/openshift/origin-release:v4.0`.
+## Troubleshooting
 
-## Running CVO tests
+A typical OpenShift cluster will have a `Deployment` resource called
+`cluster-version-operator` in the `openshift-cluster-version` namespace,
+configured to run a single CVO replica. Confirm that its `Pod` is up and optionally
+inspect its log:
 
-```sh
-# Run all unit tests
-go test ./...
+```console
+$ oc get deployment -n openshift-cluster-version cluster-version-operator
+NAME                       READY   UP-TO-DATE   AVAILABLE   AGE
+cluster-version-operator   1/1     1            1           2y227d
 
-# Run integration tests against a cluster (creates content in a given namespace)
-# Requires the CVO CRD to be installed.
-export KUBECONFIG=<admin kubeconfig>
-TEST_INTEGRATION=1 go test ./... -test.run=^TestIntegration
+$ oc get pods -n openshift-cluster-version -l k8s-app=cluster-version-operator
+NAME                                        READY   STATUS    RESTARTS   AGE
+cluster-version-operator-7fdf58f594-ct92x   1/1     Running   1          85d
+
+$ oc logs -n openshift-cluster-version -l k8s-app=cluster-version-operator
+...
 ```
 
-## Contributing
+The CVO follows the [Kubernetes API conventions](https://github.com/kubernetes/community/blob/4c9ef2d/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) and sets `Conditions` in the status of its `ClusterVersion` resource:
```

Comment:
nit: probably spell out the full commit hash (4c9ef2d135294355e7ca33ec7a5e01d31438df12) in the URI.  That reduces the risk of a truncated-hash collision as additional commits land upstream, and the user doesn't have to read this URI anyway.

Reply:
also, possibly drop this section entirely, because both `oc adm upgrade` and the web-console (which you've already plugged) will handle rendering conditions for the user.

Reply:
yes and yes, done in bf6cc77a13f199601f2f472ce21d6aa2af8d9c1c